openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/CollectionDAO.java

```diff
@@ -2657,36 +2657,152 @@ default String getNameColumn() {
     @Override
     default List<String> listBefore(ListFilter filter, int limit, String before) {
       String entityType = filter.getQueryParam("entityType");
+      String testPlatform = filter.getQueryParam("testPlatform");
       String condition = filter.getCondition();
 
-      if (entityType == null) {
+      if (entityType == null && testPlatform == null) {
         return EntityDAO.super.listBefore(filter, limit, before);
       }
-      condition = String.format("%s AND entityType='%s' ", condition, entityType);
-      return listBefore(getTableName(), getNameColumn(), condition, limit, before);
+
+      if (entityType == null) {
+        String mysqlCondition =
+            String.format("%s AND json_extract(json, '$.testPlatforms') LIKE '%%%s%%' ", condition, testPlatform);
+        String psqlCondition = String.format("%s AND json->>'testPlatforms' LIKE '%%%s%%' ", condition, testPlatform);
+        return listBefore(getTableName(), getNameColumn(), mysqlCondition, psqlCondition, limit, before);
+      }
+
+      if (testPlatform == null) {
+        condition = String.format("%s AND entityType='%s' ", condition, entityType);
+        return listBefore(getTableName(), getNameColumn(), condition, condition, limit, before);
+      }
+
+      String mysqlCondition =
+          String.format(
+              "%s AND entityType='%s' AND json_extract(json, '$.testPlatforms') LIKE '%%%s%%' ",
+              condition, entityType, testPlatform);
+      String psqlCondition =
+          String.format(
+              "%s AND entityType='%s' AND json->>'testPlatforms' LIKE '%%%s%%' ", condition, entityType, testPlatform);
+      return listBefore(getTableName(), getNameColumn(), mysqlCondition, psqlCondition, limit, before);
```

Comment:
I think we can build the SQL query differently, and this could be applied to other methods and maybe also extracted into a separate method.
```suggestion
      StringBuilder mysqlCondition = new StringBuilder();
      StringBuilder psqlCondition = new StringBuilder();

      if (entityType != null) {
        mysqlCondition.append(String.format("%s AND json_extract(json, '$.testPlatforms') LIKE '%%%s%%' ", condition, testPlatform));
        psqlCondition.append(String.format("%s AND json->>'testPlatforms' LIKE '%%%s%%' ", condition, testPlatform));
      }

      if (testPlatform != null) {
        mysqlCondition.append(String.format("%s AND entityType='%s' ", condition, entityType));
        psqlCondition.append(String.format("%s AND entityType='%s' ", condition, entityType));
      }
      
      return listBefore(getTableName(), getNameColumn(), mysqlCondition.toString(), psqlCondition.toString(), limit, before);
```